source/server/listener_manager_impl.cc

```diff
@@ -456,17 +456,23 @@ void ListenerImpl::setSocket(const Network::SocketSharedPtr& socket) {
 
 ListenerManagerImpl::ListenerManagerImpl(Instance& server,
                                          ListenerComponentFactory& listener_factory,
-                                         WorkerFactory& worker_factory)
-    : server_(server), factory_(listener_factory), stats_(generateStats(server.stats())),
+                                         WorkerFactory& worker_factory,
+                                         SystemTimeSource& system_time_source)
+    : server_(server), system_time_source_(system_time_source), factory_(listener_factory),
+      stats_(generateStats(server.stats())),
       config_tracker_entry_(server.admin().getConfigTracker().add(
           "listeners", [this] { return dumpListenerConfigs(); })) {
+
+  listener_config_update_time_ = system_time_source_.currentTime();
   for (uint32_t i = 0; i < std::max(1U, server.options().concurrency()); i++) {
     workers_.emplace_back(worker_factory.createWorker());
   }
 }
 
 ProtobufTypes::MessagePtr ListenerManagerImpl::dumpListenerConfigs() {
   auto config_dump = std::make_unique<envoy::admin::v2alpha::ListenersConfigDump>();
+  config_dump->mutable_last_updated()->MergeFrom(Protobuf::util::TimeUtil::TimeTToTimestamp(
```

Comment:
You can probably add some methods to https://github.com/envoyproxy/envoy/blob/master/source/common/protobuf/utility.h#L288 to factor this out.